Lesson1HomeTask/task1/main.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+
+using my_vec_t = std::vector<int>;
+
+my_vec_t my_merge1 (const my_vec_t& v1, const my_vec_t& v2)
+{
+    my_vec_t result {v1};
+    const auto  result_size = v1.size() + v2.size();
+    result.resize(result_size);
```

Comment:
That's a common mistake.
Reserve will only allocate new memory + copy to new place.
Resize will reserve + default initialize the "**extra**" space.
https://stackoverflow.com/questions/7397768/choice-between-vectorresize-and-vectorreserve

And also reserve is redudant here, because we'll have the next sequence:
1. Allocate memory, do copy
2. Reserve will reallocate it and do copy.

Please think how we can remove one allocation + copy in this case.

Reply:
Hmmmm...
What "reserve" you talk about? I do have "resize" method.

Reply:
> Please think how we can remove one allocation + copy in this case. 
I thought about 'emplays' but idk how to apply it here...

Reply:
Yes, but resize is not the best solution here.
Note, when you specify the size for container (it is specified by copy), it means it is reserved.

Reply:
Okay, just follow this steps:
1. Create vector with size of 1st vector + size of 2nd.
2. Copy 1st vector to created.
So, no additional reallocations are needed.